packages/node/src/integrations/localvariables.ts

```diff
@@ -0,0 +1,236 @@
+import { getCurrentHub } from '@sentry/core';
+import { Event, EventProcessor, Integration, StackFrame, StackParser } from '@sentry/types';
+import { Debugger, InspectorNotification, Runtime, Session } from 'inspector';
+import { LRUMap } from 'lru_map';
+
+import { NodeClient } from '../client';
+
+/**
+ * Promise API is available as `Experimental` and in Node 19 only.
+ *
+ * Callback-based API is `Stable` since v14 and `Experimental` since v8.
+ * Because of that, we are creating our own `AsyncSession` class.
+ *
+ * https://nodejs.org/docs/latest-v19.x/api/inspector.html#promises-api
+ * https://nodejs.org/docs/latest-v14.x/api/inspector.html
+ */
+class AsyncSession extends Session {
+  public async getProperties(objectId: string): Promise<Runtime.PropertyDescriptor[]> {
+    return new Promise((resolve, reject) => {
+      this.post(
+        'Runtime.getProperties',
+        {
+          objectId,
+          ownProperties: true,
+        },
+        (err, params) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve(params.result);
+          }
+        },
+      );
+    });
+  }
+}
+
+// Add types for the exception event data
+type PausedExceptionEvent = Debugger.PausedEventDataType & {
+  data: {
+    // This contains error.stack
+    description: string;
+  };
+};
+
+/** Could this be an anonymous function? */
+function isAnonymous(name: string | undefined): boolean {
+  return name !== undefined && ['', '?', '<anonymous>'].includes(name);
+}
+
+/** Do the function names appear to match? */
+function functionNamesMatch(a: string | undefined, b: string | undefined): boolean {
+  return a === b || (isAnonymous(a) && isAnonymous(b));
+}
+
+/** Creates a unique hash from stack frames */
+function hashFrames(frames: StackFrame[] | undefined): string | undefined {
+  if (frames === undefined) {
+    return;
+  }
+
+  return frames.reduce((acc, frame) => `${acc},${frame.function},${frame.lineno},${frame.colno}`, '');
+}
+
+interface FrameVariables {
+  function: string;
+  vars?: Record<string, unknown>;
+}
+
+/**
+ * Adds local variables to exception frames
+ */
+export class LocalVariables implements Integration {
+  public static id: string = 'LocalVariables';
+
+  public readonly name: string = LocalVariables.id;
+
+  private readonly _session: AsyncSession = new AsyncSession();
+  private readonly _cachedFrames: LRUMap<string, Promise<FrameVariables[]>> = new LRUMap(50);
+  private _stackParser: StackParser | undefined;
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(addGlobalEventProcessor: (callback: EventProcessor) => void): void {
+    const options = getCurrentHub().getClient<NodeClient>()?.getOptions();
```

Comment:
m: can we early return and log out a warning if not Node 14?

Reply:
There were a few minor issues but the integration tests are now passing for node v8.

Reply:
ah perfect! 🎉